### PR TITLE
Teachers List Connection

### PR DESCRIPTION
**Pull request by Jimmy-Tempest**
image

**Features**
- Teacher List page

### CODE REVIEW

File: src/utils/backend/person/teacher.ts [hunk 1]
```diff
@@ -5,8 +5,14 @@ import { PostgrestError } from "@supabase/supabase-js";
...
 // Types
-import { PersonDB, TeacherTable } from "@utils/types/database/person";
+import {
+  PersonDB,
+  TeacherDB,
+  TeacherTable,
+} from "@utils/types/database/person";
+import { RoomSubjectDB } from "@utils/types/database/subject";
 import { Teacher } from "@utils/types/person";
+import { db2Teacher } from "../database";
```
Comment (SiravitPhokeed) [inline src/utils/backend/person/teacher.ts:15]: You know what you need to do. 👀

File: src/utils/backend/person/teacher.ts [hunk 2]
```diff
@@ -57,3 +63,55 @@ export async function createTeacher(
...
+  const selected_teachers: (TeacherDB | null)[] = await Promise.all(
+    // 2. Flatten the arrays into an array of teacher IDs
+    roomSubjects
+      .map((roomSubject) => roomSubject.teacher)
+      .flat()
+      // remove duplicates
+      .filter((id, index, self) => self.indexOf(id) === index)
+      // 3. Fetch teacher data for each teacher
+      .map(async (teacher_id) => {
+        // TODO: Get teacher from ID (select only id, name, contacts, subject group)
```
Comment (SiravitPhokeed) [inline src/utils/backend/person/teacher.ts:90]: I'm not sure I understand these comments… (where's 1?) also remove TODOs if you're done with them.

**Review by SiravitPhokeed — APPROVED** [review]
LGTM but perhaps you should consider updating the photo to reflect the page’s current condition!

**Review by Temp9699 — APPROVED** [review]
LGTM

**Pull request merged**